src/python/google_cloud_utils/storage.py

```diff
@@ -294,7 +294,7 @@ def write_data(self, data, remote_path, metadata=None):
       blob.upload_from_string(data)
     except google.cloud.exceptions.GoogleCloudError:
       logs.log_error('Failed to write cloud storage file %s.' % remote_path)
-      return False
+      raise
```

Comment:
Since you are raising error here, should these log_error be log_warn ? and then let retry do the log_error when it completely failed ?

Reply:
yeah, good point!

Reply:
Done!